display/d.barscale/draw_scale.c

```diff
@@ -331,7 +331,7 @@ int draw_scale(double east, double north, int length, int seg, int units,
             xarr[1] = seg_len;
             yarr[1] = 0;
             xarr[2] = 0;
-            yarr[2] = (i % 2 ? -ysize_checker : ysize_checker);
+            yarr[2] = +ysize_checker;
```

Comment:
It wasn't this line that issued the warning. No need to change.

Reply:
Sorry for the confusion I have force pushed the correct code